### PR TITLE
Implement story 317

### PR DESCRIPTION
**Pull request by cewing**
This PR implements the 'allow_guests' functionality for work party types.

If a work party type does not allow guests, then anonymous visitors to VMS will be forwarded to My Backpack to log in or register before they are sent on to the registration page for the work party.

In addition, anonymous users will see a text description of this when viewing a work party that does not allow guest registrations.

### CODE REVIEW

**Comment by cewing** [comment]
Py2 build is working, Py3 still failing as expected due to the intelligent text package issue (known issue)

File: src/vms/static/app/workparty.js [hunk 1]
```diff
@@ -215,6 +216,7 @@ var WorkPartyStatus = module.exports.WorkPartyStatus = React.createClass({
 			);
 		} else {
 			var register_action = wp.actions.register;
+			var guest_warning = allow_guests ? '' : <h3>A My Backpack account is required. You will be asked to register or log in before you can join.</h3>;
```
Comment (davisagli) [inline src/vms/static/app/workparty.js]: We should only show this message for logged-out users.

**Comment by davisagli** [comment]
Needs one fix; otherwise looks good so feel free to merge after that's fixed.

**Comment by cewing** [comment]
Thanks for the catch, @davisagli.  Had that in the pyramid template but missed it here.

Fixed.  Merging now.

**Pull request merged**